src/VIPServer.java

```diff
@@ -107,24 +107,6 @@ public void main(String[] args) throws HostFailureException, HostNotFoundExcepti
 					// SHOULD BE REPLACED BY
 					// job.carryOn();
 				} else {
-					if (!timer) {
```

Comment:
You should not remove this block. Check with Sorina but it corresponds to what is done by the real system. It aims at avoiding to wait for stragglers among worker nodes and trigger the merge even if some workers are not done.